src/app/static/src/app/components/adr/SelectDatasetModal.vue

```diff
@@ -0,0 +1,130 @@
+            <hint-tree-select id="datasetSelector"
+                              :multiple="false"
+                              :searchable="true"
+                              :options="datasetOptions"
+                              :placeholder="select"
+                              :disabled="fetchingDatasets"
+                              v-model="datasetId">
+                <template v-slot:option-label="{node}">
+                    <label v-html="node.raw.customLabel">
+                    </label>
+                </template>
+            </hint-tree-select>
+            <select-release :dataset-id="datasetId ? datasetId : null"
+                            :open="open"
+                            :dataset-type="datasetType"
+                            @selected-dataset-release="(newReleaseId) => datasetReleaseId = newReleaseId"
+                            @valid="(newValue) => valid = newValue">
+            </select-release>
+            <div :class="fetchingDatasets ? 'visible' : 'invisible'"
+                 style="margin-top: 15px"
+                 id="fetching-datasets">
+                <loading-spinner size="xs"></loading-spinner>
+                <span v-translate="'loadingDatasets'"></span>
+            </div>
+            <div v-if="fetchingError" id="fetch-error">
+                <div v-if="fetchingError.detail">{{fetchingError.detail}}</div>
+                <button @click="fetchDatasets"
+                        class="btn btn-red float-right"
+                        v-translate="'tryAgain'">
+                </button>
+            </div>
+        </div>
+        <div class="text-center" v-if="loading" id="loading-dataset">
+            <loading-spinner size="sm"></loading-spinner>
+        </div>
+        <template v-slot:footer v-if="!loading">
+            <button id="importBtn"
+                    type="button"
+                    :disabled="disableImport"
+                    class="btn btn-red"
+                    v-translate="'import'"
+                    @click="$emit('confirmImport', datasetId, datasetReleaseId)">
+            </button>
+            <button type="button"
+                    class="btn btn-white"
+                    v-translate="'cancel'"
+                    @click="$emit('closeModal')">
+            </button>
+        </template>
+    </modal>
+</template>
+
+<script setup lang="ts">
+import HintTreeSelect from "../HintTreeSelect.vue";
+import LoadingSpinner from "../LoadingSpinner.vue";
+import SelectRelease from "./SelectRelease.vue";
+import Modal from "../Modal.vue";
+import {useStore} from "vuex";
+import {RootState} from "../../root";
+import i18next from "i18next";
+import {computed, PropType, ref, watch, watchEffect} from "vue";
+import {AdrDatasetType} from "../../store/adr/adr";
+import {Release} from "../../types";
+
+// eslint-disable-next-line no-undef
+defineEmits<{
+    (e: "confirmImport", datasetId: string, release: Release): void
```

Comment:
I believe there may be a type mismatch here. It looks like the confirmImport method on the ADRInputDataset component is expecting a releaseId as string for the second param, where the custom event definition defines the second param as type "Release".